test/topology_custom/test_tablets.py

```diff
@@ -45,6 +45,21 @@ async def test_tablet_replication_factor_enough_nodes(manager: ManagerClient):
     await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int);")
 
 
+@pytest.mark.asyncio
+async def test_tablet_scaling_option_is_respected(manager: ManagerClient):
+    # 32 is high enough to ensure we demand more tablets than the default choice.
+    cfg = {'enable_tablets': True, 'tablets_initial_scale_factor': 32}
+    servers = await manager.servers_add(1, config=cfg, cmdline=['--smp', '2'])
+
+    cql = manager.get_cql()
+
+    await cql.run_async(f"CREATE KEYSPACE test WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}")
```

Comment:
If this PR lands after https://github.com/scylladb/scylladb/pull/22399, please use `async with new_test_keyspace`.